printer.go

```diff
@@ -0,0 +1,181 @@
+package kubeaudit
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/Shopify/kubeaudit/internal/color"
+	"github.com/Shopify/kubeaudit/internal/k8s"
+	"github.com/Shopify/kubeaudit/k8stypes"
+	log "github.com/sirupsen/logrus"
+)
+
+type Printer struct {
+	writer      io.Writer
+	minSeverity SeverityLevel
+	formatter   log.Formatter
+	color       bool
+}
+
+type PrintOption func(p *Printer)
+
+func WithMinSeverity(minSeverity SeverityLevel) PrintOption {
+	return func(p *Printer) {
+		p.minSeverity = minSeverity
+	}
+}
+
+func WithWriter(writer io.Writer) PrintOption {
+	return func(p *Printer) {
+		p.writer = writer
+	}
+}
+
+func WithFormatter(formatter log.Formatter) PrintOption {
+	return func(p *Printer) {
+		p.formatter = formatter
+	}
+}
+
+func (p *Printer) parseOptions(opts ...PrintOption) {
+	for _, opt := range opts {
+		opt(p)
+	}
+}
+
+func NewPrinter(opts ...PrintOption) Printer {
+	p := Printer{
+		writer:      os.Stdout,
+		minSeverity: Info,
+	}
+	p.parseOptions(opts...)
+	if p.writer == os.Stdout {
+		p.color = true
+	}
+	return p
+}
+
+func (p *Printer) PrintReport(report *Report) {
+	if p.formatter == nil {
+		p.prettyPrintReport(report)
+	} else {
+		p.logReport(report)
+	}
+}
+
+func (p *Printer) prettyPrintReport(report *Report) {
+	for _, workloadResult := range report.ResultsWithMinSeverity(p.minSeverity) {
+		resource := workloadResult.GetResource().Object()
+		groupVersionKind := resource.GetObjectKind().GroupVersionKind()
+		resourceName := k8s.GetObjectMeta(resource).GetName()
+		resourceNamespace := k8s.GetObjectMeta(resource).GetNamespace()
+
+		p.printColor(color.CyanColor, "\n--------- Results for ---------------------\n\n")
```

Comment:
nitpick! can we me make it more centralized? 😬

`"\n---------------- Results for ---------------\n\n"
`
(I counted so it gets centralized like this. 17 to the left, 17 to the right)